src/alloc.c

```diff
@@ -438,12 +437,7 @@ static jl_sym_t *symtab_lookup(jl_sym_t *volatile *ptree, const char *str,
             ptree = &node->left;
         else
             ptree = &node->right;
-        node = *ptree;
-        // We may need a data dependency barrier here to pair with the
-        // cpu write barrier in _jl_symbol so that we won't read a invalid
-        // value from `node`. However, this shouldn't be a problem on all
-        // the architectures we currently support.
-        // https://www.kernel.org/doc/Documentation/memory-barriers.txt
+        node = jl_atomic_load_acquire(ptree);
```

Comment:
Another note is that I think this one can probably be a `load_consume`? Since it is synchronizing the load of the next object and the load from that object.

However, given that (AFAIK) this doesn't really make a difference on x86 and currently no compiler actually implement the weaker version, I can't really test and verify that it actually works....